onnxruntime/core/optimizer/transformer_memcpy.cc

```diff
@@ -150,15 +153,19 @@ bool TransformerMemcpyImpl::ModifyGraph(const KernelRegistryManager& kernel_regi
   return modified;
 }
 
-void TransformerMemcpyImpl::ProcessDefs(onnxruntime::Node& node, const KernelRegistryManager& kernel_registries) {
+void TransformerMemcpyImpl::ProcessDefs(onnxruntime::Node& node, const KernelRegistryManager& kernel_registries, InitializedTensorSet& initializers_scoped_to_current_graph_level) {
   if (node.GetExecutionProviderType() == provider_) {
     provider_nodes_.insert(&node);
     // note KernelCreateInfo might be nullptr for custom kernel
     const KernelCreateInfo* kci = nullptr;
     kernel_registries.SearchKernelRegistry(node, &kci);
 
     auto status = onnxruntime::Node::ForEachWithIndex(node.InputDefs(),
-                                                      [this, &kci](const onnxruntime::NodeArg& arg, size_t index) {
+                                                      [this, &kci, &initializers_scoped_to_current_graph_level](const onnxruntime::NodeArg& arg, size_t index) {
+                                                        // check if this NodeArg corresponds to an initializer defined in current graph or outer graph level 
+                                                        const auto* initializer_tensor_proto = graph_utils::ReturnInitializedValue(graph_, arg.Name(), true);
+                                                        if (initializer_tensor_proto != nullptr)
+                                                          initializers_scoped_to_current_graph_level[arg.Name()] = initializer_tensor_proto;
```

Comment:
>initializers_scoped_to_current_graph_level [](start = 58, length = 42)

name seems slightly confusing when you can match initializers from outer scope. would something like 'initializers_consumed' be better?

Reply:
Definitely sounds better. Will change it.